### ISSUE TITLE
orphan section `__cpuidle_method_of_table'

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#1372]
via: https://lists.01.org/hyperkitty/list/kbuild-all@lists.01.org/thread/KV3K4HNCDQ54SP6MJQNF63WAGFPWV6IT/

```
$ wget https://lists.01.org/hyperkitty/list/kbuild-all@lists.01.org/message/KV3K4HNCDQ54SP6MJQNF63WAGFPWV6IT/attachment/2/config.gz -O - | gunzip > .config
$ ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- make LLVM=1 LLVM_IAS=1 -j72 -s olddefconfig all
...
ld.lld: warning: arch/arm/mach-omap2/built-in.a(pm33xx-core.o):(__cpuidle_method_of_table) is being placed in '__cpuidle_method_of_table'
ld.lld: warning: arch/arm/mach-omap2/built-in.a(pm33xx-core.o):(__cpuidle_method_of_table) is being placed in '__cpuidle_method_of_table'
ld.lld: warning: arch/arm/mach-omap2/built-in.a(pm33xx-core.o):(__cpuidle_method_of_table) is being placed in '__cpuidle_method_of_table'
```
via `CPUIDLE_METHOD_OF_DECLARE`.

We could put `__cpuidle_method_of_table` in `RODATA`, but I'm curious if this section is really necessary?
```diff
diff --git a/include/asm-generic/vmlinux.lds.h b/include/asm-generic/vmlinux.lds.h
index 17325416e2de..92c6d9eb0fc4 100644
--- a/include/asm-generic/vmlinux.lds.h
+++ b/include/asm-generic/vmlinux.lds.h
@@ -433,6 +433,7 @@
        .rodata           : AT(ADDR(.rodata) - LOAD_OFFSET) {           \
                __start_rodata = .;                                     \
                *(.rodata) *(.rodata.*)                                 \
+               *(__cpuidle_method_of_table)                            \
                SCHED_DATA                                              \
                RO_AFTER_INIT_DATA      /* Read only after init */      \
                . = ALIGN(8);                                           \
```

**Comment by nathanchance** [comment]
This has a patch from Arnd that has reviews from you and Miguel, it needs chasing I guess?

https://lore.kernel.org/r/20201230155506.1085689-1-arnd@kernel.org

https://lore.kernel.org/r/202104071241.DDD60D1B49@keescook/

**Comment by kees** [comment]
I'll snag this...

**Comment by nickdesaulniers** [comment]
accepted: https://git.kernel.org/pub/scm/linux/kernel/git/kees/linux.git/commit/?h=for-next/orphans&id=d94b93a9101573eb75b819dee94b1417acff631b

**Comment by nathanchance** [comment]
Merged into mainline: https://git.kernel.org/linus/d94b93a9101573eb75b819dee94b1417acff631b
